File: JavaScript/Math-Array-String/58.js
// 1313. Decompress Run-Length Encoded List


/**
 * @param {number[]} nums
 * @return {number[]}
 */
var decompressRLElist = function(nums) {
    let result = [];
    for (let i = 0; i < nums.length; i += 2) {
        let freq = nums[i];
        let val = nums[i + 1];
        result.push(...Array(freq).fill(val));
    }
    return result;
};

// Example usage:
let nums1 = [1, 2, 3, 4];
let nums2 = [1, 1, 2, 3];

console.log(decompressRLElist(nums1)); // Output: [2, 4, 4, 4]
console.log(decompressRLElist(nums2)); // Output: [1, 3, 3]

File: Python/Math-Array-String/58.py
# 1313. Decompress Run-Length Encoded List

from typing import List

class Solution:
    def decompressRLElist(self, nums: List[int]) -> List[int]:
        result = []
        for i in range(0, len(nums), 2):
            freq = nums[i]
            val = nums[i + 1]
            result.extend([val] * freq)
        return result
    

sol = Solution()
output = sol.decompressRLElist([1, 2, 3, 4])
print(output) # # Output: [2, 4, 4, 4]